Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java

```diff
@@ -229,60 +229,41 @@ public static boolean enableAllowAccessFromLocalMachine(final String subscriptio
             if (MySQLMvpModel.FirewallRuleMvpModel.isAllowAccessFromLocalMachine(subscriptionId, server)) {
                 return true;
             }
-            try {
-                final String publicIp = MySQLMvpModel.FirewallRuleMvpModel.getPublicIp();
-                final String ruleName = NAME_PREFIX_ALLOW_ACCESS_TO_LOCAL + publicIp.replaceAll("\\.", "-");
-                final FirewallRuleInner firewallRule = new FirewallRuleInner();
-                firewallRule.withStartIpAddress(publicIp);
-                firewallRule.withEndIpAddress(publicIp);
-                final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
-                mySQLManager.firewallRules().inner().createOrUpdate(server.resourceGroupName(), server.name(), ruleName, firewallRule);
-            } catch (IOException e) {
-                Lombok.sneakyThrow(e);
+            MySQLConnectionUtils.ConnectResult connectResult = MySQLConnectionUtils.connectWithPing(JdbcUrl.mysql(server.fullyQualifiedDomainName()),
+                    server.administratorLogin() + "@" + server.name(), StringUtils.EMPTY);
+            if (StringUtils.isNotBlank(connectResult.getMessage())) {
+                Matcher matcher = IPADDRESS_PATTERN.matcher(connectResult.getMessage());
+                if (matcher.find()) {
+                    final String publicIp = matcher.group();
+                    final String ruleName = MySQLMvpModel.FirewallRuleMvpModel.getAllAccessFromLocalRuleName();
+                    final FirewallRuleInner firewallRule = new FirewallRuleInner();
+                    firewallRule.withStartIpAddress(publicIp);
+                    firewallRule.withEndIpAddress(publicIp);
+                    final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+                    mySQLManager.firewallRules().inner().createOrUpdate(server.resourceGroupName(), server.name(), ruleName, firewallRule);
+                    return true;
```

Comment:
this method actually does 3 tasks. 
1. ip = getLocalIp.
2. rule = buildLocalFirewallRule(ip)
3. addLocalIp2Firewall(rule)

Reply:
yes. so the name of this method is "enableAllowAccessFromLocalMachine"

Reply:
I mean please refactor this method to extract at least the getLocalIp method, so that we can just update the getLocalIp method when we have a better way to get local ip.